pkg/parser/model/ddl.go

```diff
@@ -180,9 +182,177 @@ var actionMap = map[ActionType]string{
+	ActionAlterTablePartitionAttributes: UnsafeDDL,
+	ActionCreatePlacementPolicy:         UnmanagementDDL,
+	ActionAlterPlacementPolicy:          UnmanagementDDL,
+	ActionDropPlacementPolicy:           UnmanagementDDL,
+	ActionAlterTablePartitionPlacement:  UnsafeDDL,
+	ActionModifySchemaDefaultPlacement:  UnsafeDDL,
+	ActionAlterTablePlacement:           UnsafeDDL,
+	ActionAlterCacheTable:               UnsafeDDL,
+	ActionAlterTableStatsOptions:        UnsafeDDL,
+	ActionAlterNoCacheTable:             UnsafeDDL,
+	ActionCreateTables:                  SafeDDL,
+	ActionMultiSchemaChange:             UnsafeDDL,
+	ActionFlashbackCluster:              UnsafeDDL,
+	ActionRecoverSchema:                 UnsafeDDL,
+	ActionReorganizePartition:           UnsafeDDL,
+	ActionAlterTTLInfo:                  SafeDDL,
+	ActionAlterTTLRemove:                SafeDDL,
+	ActionCreateResourceGroup:           UnmanagementDDL,
+	ActionAlterResourceGroup:            UnmanagementDDL,
+	ActionDropResourceGroup:             UnmanagementDDL,
+	ActionAlterTablePartitioning:        UnsafeDDL,
+	ActionRemovePartitioning:            UnsafeDDL,
+}
+
+// BDRActionMap is the map of DDLBDRType to ActionType (reversed from ActionBDRMap).
+var BDRActionMap = map[DDLBDRType][]ActionType{
+	SafeDDL: {
+		ActionCreateSchema,
+		ActionCreateTable,
+		ActionAddColumn,
+		ActionAddIndex,
+		ActionDropIndex,
+		ActionModifyColumn,
+		ActionSetDefaultValue,
+		ActionModifyTableComment,
+		ActionRenameIndex,
+		ActionAddTablePartition,
+		ActionDropPrimaryKey,
+		ActionAlterIndexVisibility,
+		ActionCreateTables,
+		ActionAlterTTLInfo,
+		ActionAlterTTLRemove,
+	},
+	UnsafeDDL: {
+		ActionDropSchema,
+		ActionDropTable,
+		ActionDropColumn,
+		ActionAddForeignKey,
+		ActionDropForeignKey,
+		ActionTruncateTable,
+		ActionRebaseAutoID,
+		ActionRenameTable,
+		ActionShardRowID,
+		ActionDropTablePartition,
+		ActionModifyTableCharsetAndCollate,
+		ActionTruncateTablePartition,
+		ActionRecoverTable,
+		ActionModifySchemaCharsetAndCollate,
+		ActionLockTable,
+		ActionUnlockTable,
+		ActionRepairTable,
+		ActionSetTiFlashReplica,
+		ActionUpdateTiFlashReplicaStatus,
+		ActionAddPrimaryKey,
+		ActionCreateSequence,
+		ActionAlterSequence,
+		ActionDropSequence,
+		ActionModifyTableAutoIdCache,
+		ActionRebaseAutoRandomBase,
+		ActionExchangeTablePartition,
+		ActionAddCheckConstraint,
+		ActionDropCheckConstraint,
+		ActionAlterCheckConstraint,
+		ActionRenameTables,
+		ActionAlterTableAttributes,
+		ActionAlterTablePartitionAttributes,
+		ActionAlterTablePartitionPlacement,
+		ActionModifySchemaDefaultPlacement,
+		ActionAlterTablePlacement,
+		ActionAlterCacheTable,
+		ActionAlterTableStatsOptions,
+		ActionAlterNoCacheTable,
+		ActionMultiSchemaChange,
+		ActionFlashbackCluster,
+		ActionRecoverSchema,
+		ActionReorganizePartition,
+		ActionAlterTablePartitioning,
+		ActionRemovePartitioning,
+	},
+	UnmanagementDDL: {
+		ActionCreateView,
+		ActionDropView,
```

Comment:
Should they be classified into safe DDL? @asddongmen 

<img width="794" alt="image" src="https://github.com/pingcap/tidb/assets/23556834/f4cff9f8-82a7-40e5-849f-ebbdd7ec88d7">

Reply:
yes, it is safe DDL